src/track/beatiterator.h

```diff
@@ -23,23 +22,21 @@ class BeatIterator final {
     using iterator_category = std::forward_iterator_tag;
 
     bool hasNext() const {
-        return !m_ended;
+        return m_currentBeat != m_endBeat;
     }
 
-    /// Advances de iterator and returns the current beat frame position.
+    /// Advances the iterator and returns the current beat frame position.
     /// If you need the frame position make sure you store it, is not possible
     /// to get it again.
     double next() {
```

Comment:
Should this return a FramePos or a Beat?

Reply:
FramePos here. But I am returning an object of Beat class in #2844